drivers/nrf24l01p/nrf24l01p_communication.c

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (C) 2019 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+/**
+ * @ingroup drivers_nrf24l01p
+ * @{
+ *
+ * @file
+ * @brief     Functions to communicate with the NRF24L01P transceiver via SPI
+ *
+ * @author Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author Peter Kietzmann <peter.kietzmann@haw-hamburg.de>
+ * @author Fabian Hüßler <fabian.huessler@ovgu.de>
+ * @}
+ */
+#include <errno.h>
+#include <string.h>
+
+#include "periph/gpio.h"
+#include "periph/spi.h"
+#include "xtimer.h"
+#include "nrf24l01p_constants.h"
+#include "nrf24l01p_communication.h"
+
+#define ENABLE_DEBUG    (0)
+
+#ifndef MIN
+#define MIN(a, b) (((a) < (b)) ? (a) : (b))
+#endif
+
+#define SPI_BUS     (dev->params.spi)
+#define SPI_PIN_CS  (dev->params.pin_cs)
+#define DUMMY       NULL
+
+#define NRF24L01P_REG_MAX_WIDTH NRF24L01P_MAX_ADDR_WIDTH
+
+static void swap_bytes_inplace(void *buffer, size_t len)
+{
+    uint8_t *bytes = (uint8_t *)buffer;
+
+    for (size_t i = 0; i < len / 2; i++) {
+        uint8_t tmp = bytes[i];
+        bytes[i] = bytes[len - 1 - i];
+        bytes[len - 1 - i] = tmp;
+    }
+}
```

Comment:
Is this for conversion between host and network byte order? If so, could one of the standard C functions (`htonl()` and friends) or one of those in [`byteorder.h`](http://api.riot-os.org/byteorder_8h.html) be used instead?

Reply:
If you write multiple bytes e.g. payload to the `nrf24l01p` it expects them in reverse order.
It is not for the byteorder of numerical values.